<launch>
    <arg name="model" default='$(find anrobot)/urdf/movable.urdf' />
    <arg name="gui" default="False" />
    <arg name="use_lin" default="True" />

    <param name="robot_description" textfile="$(arg model)" />
    <param name="use_gui" value="$(arg gui)"/>
    <param name="use_lin" value="$(arg use_lin)"/>

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" >
        <rosparam>
              zeros:
                joint1: 0.1
                joint2: 0.3
                joint3: 1.5
        </rosparam>
    </node>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" >
        <remap from="joint_states" to="trajectory_joint_states" />
    </node>
    <node name="inv_gen_trajectory" pkg="anrobot" type="inv_gen_trajectory" />
    <node name="inv_kinematics" pkg="anrobot" type="inv_kinematics" />
    <node name="trajectory_marker_pub" pkg="anrobot" type="trajectory_marker_pub" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find anrobot)/rviz/urdf.rviz" required="true" />
    <node pkg="rosservice" type="rosservice" name="pos_init" args="call /set_end_target &quot;point: {x: 1.0, y: 2.0, z: -1.0}&quot;"/>
</launch>

